ocean_lib/common/ddo/ddo.py

```diff
@@ -354,3 +387,64 @@ def create_authentication_from_json(values):
 
         authentication = {"type": authentication_type, "publicKey": key_id}
         return authentication
+
+    def get_addresses_of_type(self, section_type="allow"):
+        """Get a filtered list of addresses from credentials (use with allow/deny)."""
+        entries = self._credentials.get(section_type, [])
+
+        for entry in entries:
+            if entry["type"] == "address":
+                return [val.lower() for val in entry["values"]]
+
+        return []
+
+    def enable(self):
+        if not self._status:
+            self._status = {}
+
+        self._status.pop("isOrderDisabled")
```

Comment:
I went for implicit rather than explicit. If that key is not present, I think the structure of the DDO is clearer, otherwise a lot of flags will be set even if not needed. What do you think?

Reply:
I went for implicit rather than explicit. If that key is not present, I think the structure of the DDO is clearer, otherwise a lot of flags will be set even if not needed. What do you think?